internal/dataplane/parser/translate_httproute.go

```diff
@@ -255,5 +263,31 @@ func generateKongServiceFromHTTPRouteBackendRef(result *ingressRules, httproute
 		}
 	}
 
-	return service
+	if len(backendRefs) > 1 {
+		redundantBackendRefs := backendRefs[1:]
+		redundantServices := make([]*corev1.Service, len(redundantBackendRefs))
+		for i, backendRef := range redundantBackendRefs {
```

Comment:
We need additional checks in this loop (or rather, in a similar loop inside the admission controller) against compatibility with the first Service, to ensure that https://docs.konghq.com/kubernetes-ingress-controller/2.2.x/references/annotations/ and/or KongIngress configuration does not contradict one another.

I don't see anything where a merge would be possible (you could merge plugin sets, but that sounds like footgun hell UX, so don't), so this can be a simple "are they identical" check: loop through all Kong annotations, confirm that they have the same value, and if not, reject it. IMO we should be strict and not even allow bundles where one Service has an annotation and the others don't have it set.

`ingress.kubernetes.io/service-upstream` is an exception, because we can allow bundles where one member of the upstream target set uses its service hostname and another uses its endpoint IPs. However, I can't really see why you'd ever want to do this, so it's maybe something we shouldn't bother supporting unless requested.

Reply:
Good call, I'll get this added.